ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -276,7 +276,9 @@ Control {
                                                   maxSafeValue, d.inputSymbol,
                                                   { noSymbol: !amountToSendInput.inputIsFiat })
 
-                visible: d.isSelectedHoldingValidAsset && root.swapSide === SwapInputPanel.SwapSide.Pay
+                // FIXME: This should be enabled after #15709 is resolved
+                // visible: d.isSelectedHoldingValidAsset && root.swapSide === SwapInputPanel.SwapSide.Pay
```

Comment:
We don't want to disable it in SwapModal iirc,

Reply:
@dlipicar

Reply:
So... Had a chat with @saledjenic .
- Setting the max amount for ERC20 tokens will always work fine, since fees get taken from the ETH balance.
- Setting the max amount for ETH might be flaky... Depending on gas costs, the calculation we do might leave enough ETH for gas or not. Moreover, the user will most likely keep some ETH balance after the transaction, which wasn't their intention when they pressed the Max button. So, overall a non-optimal UX.

For these reasons, we consider the best thing to do for 2.30 is:
- Hide/Disable the Max button when ETH is selected
- Keep it as is when an ERC20 token is selected

Then, 2.31 will bring an improved logic that will let us restore some sort of "Max" function for ETH.

@caybro @Khushboo-dev-cpp any comments?

Reply:
I feel disabling (graying out) the Max button for ETH might be better than hiding it, it would at least serve as a balance indicator

Reply:
I'd keep it (for all); if I press "MAX" something, I'm fine with spending all my money I guess? Worst case we inform the user they have not enough for the fees, so they will naturally lower the amount manually

Reply:
We have an issue to properly introduce that Max button in 2.31. 

The reason why I would hide it for ETH only is that we're artificially estimating how much lower we should go than the max balance on that chain is, in order to execute the tx instead of calculating that value. That results in having much more on the account (it's not cleaned fully) then it would be if we calculate that.

Also we don't have fees updating mechanism yet and if user waits for some time after opening send button, while fees go up, that tx will fail for sure.

Reply:
hmm if the issue is that we are not able to calculate max correctly I think we should just display max balance and remove current calculations and disable interaction. But that seeing max is quite helpful I think.

Reply:
It would be weird if the max button is disabled for ETH and not for other tokens; users would wonder why they can't click it under some circumstances

Reply:
They will wonder till the next release ;)
But also can wonder, I said that I want to spend Max (all I have) but there is something left.
Ok, then we can hide it in all cases.

Reply:
@dlipicar 
so we will disable it only for eth in sendModal, but swapModel is untouched, correct?

Reply:
> They will wonder till the next release ;) But also can wonder, I said that I want to spend Max (all I have) but there is something left. Ok, then we can hide it in all cases.

I agree, from a UX perspective, it should be disabled for all tokens not to cause confusion.